api.go

```diff
@@ -223,47 +231,53 @@ func NewRaft(conf *Config, fsm FSM, logs LogStore, stable StableStore, snaps Sna
 		logger = log.New(conf.LogOutput, "", log.LstdFlags)
 	}
 
-	// Try to restore the current term
+	// Try to restore the current term.
 	currentTerm, err := stable.GetUint64(keyCurrentTerm)
 	if err != nil && err.Error() != "not found" {
 		return nil, fmt.Errorf("failed to load current term: %v", err)
 	}
 
-	// Read the last log value
+	// Read the last log value.
 	lastIdx, err := logs.LastIndex()
 	if err != nil {
 		return nil, fmt.Errorf("failed to find last log: %v", err)
 	}
 
-	// Get the log
+	// Get the log.
 	var lastLog Log
 	if lastIdx > 0 {
 		if err = logs.GetLog(lastIdx, &lastLog); err != nil {
 			return nil, fmt.Errorf("failed to get last log: %v", err)
 		}
 	}
 
+	// Make sure we have a valid server address and ID.
+	protocolVersion := conf.ProtocolVersion
 	localAddr := ServerAddress(trans.LocalAddr())
 	localID := conf.LocalID
-	if localID == "" {
-		logger.Printf("[WARN] raft: No server ID given, using network address: %v. This default will be removed in the future. Set server ID explicitly in config.",
+	if protocolVersion < 1 || localID == "" {
+		// During the transition to the new ID system, keep this as an
+		// INFO level message. Once the new scheme has been out for a
```

Comment:
Can you comment this as `COMPAT` and a `TODO` so it's easily grepable that there is a pending action to enable a depreciate `WARN` message?